### PR TITLE
Settings fixes

### PR DESCRIPTION
**Pull request by avoidr**
This change is [Reviewable](https://reviewable.io/reviews/utox/utox/1480)

### CODE REVIEW

**Comment by avoidr** [comment]
I'm good here. Just reworded a commit message.

File: src/settings.h [hunk 1]
```diff
@@ -76,7 +68,14 @@ typedef struct utox_settings {
...
 
-    uint8_t proxy_ip[];
+    // Low level settings (network, profile, portable-mode)
+    bool     portable_mode;
+    bool     disableudp;
+    bool     enableipv6;
+    bool     proxyenable;
+    bool     force_proxy;
+    uint16_t proxy_port;
+    uint8_t proxy_ip[255]; /* coincides with TOX_MAX_HOSTNAME_LENGTH from toxcore */
```
Comment (redmanmale) [inline src/settings.h:78]: Make it `char` and all casts will go away.
And vertical alignment.

**Comment by avoidr** [comment]
> +    uint8_t proxy_ip[255]; /* coincides with TOX_MAX_HOSTNAME_LENGTH from toxcore */
> 
> Make it `char` and all casts will go away.

I was thinking about that, but `char` is misleading. The data that goes in there is not ASCII, it's potentially Unicode. `proxy_ip` is not just an IP address, it may be a hostname. `proxy_ip` needs to be renamed to `proxy_host`, like it is in toxcore, too.

> And vertical alignment.
Thanks!

**Comment by avoidr** [comment]
Rebased.

**Comment by avoidr** [comment]
Are you fine with this PR?

**Pull request merged**